rebasehelper/tests/functional/test_rebase.py

```diff
@@ -90,16 +91,22 @@ def test_rebase(self, buildtool, favor_on_conflict):
         patch = unidiff.PatchSet.from_filename(changes, encoding='UTF-8')
 
         if favor_on_conflict == 'upstream':
-            backported_patch, conflicting_patch, spec_file = patch
+            backported_patch, conflicting_patch, renamed_patch, spec_file = patch
             assert conflicting_patch.is_removed_file  # conflicting.patch
         elif favor_on_conflict == 'downstream':
-            backported_patch, conflicting_patch, spec_file = patch
+            backported_patch, conflicting_patch, renamed_patch, spec_file = patch
             assert conflicting_patch.is_modified_file  # conflicting.patch
         else:
-            backported_patch, spec_file = patch
+            backported_patch, renamed_patch, spec_file = patch
             # Non interactive mode - inapplicable patches are only commented out.
             assert [h for h in spec_file if '+#Patch1:         conflicting.patch\n' in h.target]
             assert [h for h in spec_file if '+#%%patch1 -p1\n' in h.target]
+        assert renamed_patch.is_rename  # renamed patch 0.1.patch to 0.2.patch
+        assert os.path.basename(renamed_patch.source_file) == 'renamed-0.1.patch'
+        assert os.path.basename(renamed_patch.target_file) == 'renamed-0.2.patch'
+        # Check that the renamed patch path is unchanged
+        assert [h for h in spec_file if ' Patch3:         renamed-%{version}.patch\n' in h.source]
+        assert [h for h in spec_file if ' Patch3:         renamed-%{version}.patch\n' in h.target]
```

Comment:
I don't like this very much, I think it's fragile and can easily break if the generated diff changes (you are checking for something in the context of the diff). On the other hand, I think this check makes sense, and I can't think oif any better solution.

Reply:
Good point, checking unchanged lines in a diff isn't very robust. Perhaps we could instead check that no such line was added or removed in the whole diff? So gather lines `+Patch3: ... ` and `-Patch3: ...` and check if it is an empty list, does this make more sense? The purpose of this check should be the same

Reply:
Yes, that's probably better.

Reply:
Fixed - in the end checking for Patch removal in source of the hunk should be sufficient

Reply:
LGTM.